engines/scumm/actor.cpp

```diff
@@ -2209,18 +2209,35 @@ void ScummEngine_v90he::processActors() {
 
 // Used in Scumm v8, to allow the verb coin to be drawn over the inventory
 // chest. I'm assuming that draw order won't matter here.
-void ScummEngine::processUpperActors() {
+int ScummEngine::processUpperActors() {
 	int i;
+	_upperActorQueuePos = 0;
 
+	_processing_upper_actors = true;
 	for (i = 1; i < _numActors; i++) {
 		if (_actors[i]->isInCurrentRoom() && _actors[i]->_costume && _actors[i]->_layer < 0) {
-			_actors[i]->drawActorCostume();
+			_actors[i]->drawActorCostume(false, &_virtscr[kVerbVirtScreen]);
 			_actors[i]->animateCostume();
 		}
 	}
+	_processing_upper_actors = false;
+
+	return _upperActorQueuePos;
+}
+
+void ScummEngine::removeUpperActors() {
+	int i;
+
+	for (i = 0; i < _upperActorQueuePos; i++) {
```

Comment:
Please write `for (int i = 0;...` for reducing scope of the variable.

Reply:
Got it, I merely went with this approach because `processUpperActors` used it, and I was trying to stay consistent with what was already in there. :)